core/templates/components/button-directives/create-activity-button.component.spec.ts

```diff
@@ -0,0 +1,291 @@
+    getUsername: () => 'username1',
+    getEmail: () => 'tester@example.org',
+    isLoggedIn: () => true
+  };
+
+  beforeEach(waitForAsync(() => {
+    windowRef = new MockWindowRef();
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      declarations: [
+        CreateActivityButtonComponent,
+        TranslatePipe
+      ],
+      providers: [
+        UserService,
+        {
+          provide: UrlService,
+          useClass: MockUrlService
+        },
+        {
+          provide: WindowRef,
+          useValue: windowRef
+        },
+        ExplorationCreationService
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CreateActivityButtonComponent);
+    component = fixture.componentInstance;
+    userService = TestBed.inject(UserService);
+    urlService = TestBed.inject(UrlService);
+    explorationCreationService = TestBed.inject(ExplorationCreationService);
+    ngbModal = TestBed.inject(NgbModal);
+    fixture.detectChanges();
+  });
+
+  it('should successfully instantiate the component from beforeEach block',
+    () => {
+      expect(component).toBeDefined();
+    });
+
+  it('should begin activity creation process if user can' +
+    ' create collections', fakeAsync(() => {
+    spyOn(userService, 'getUserInfoAsync')
+      .and.returnValue(Promise.resolve(userInfoForCollectionCreator));
+
+    spyOn(component, 'initCreationProcess');
+
+    expect(component.canCreateCollections).toBe(false);
+    expect(component.userIsLoggedIn).toBe(false);
+
+    component.ngOnInit();
+    tick();
+    fixture.detectChanges();
+
+    expect(component.initCreationProcess).toHaveBeenCalled();
+    expect(component.canCreateCollections).toBe(true);
+    expect(component.userIsLoggedIn).toBe(true);
+  }));
+
+  it('should create a new exploration automatically if the user' +
+    ' cannot create collections', fakeAsync(() => {
+    spyOn(userService, 'getUserInfoAsync')
+      .and.returnValue(Promise.resolve(userInfoForNonCollectionCreator));
+    const explorationCreationServiceSpy = spyOn(
+      explorationCreationService, 'createNewExploration');
+
+    expect(component.canCreateCollections).toBe(false);
+    expect(component.userIsLoggedIn).toBe(false);
+
+    component.ngOnInit();
+    tick();
+    fixture.detectChanges();
+
+    expect(explorationCreationServiceSpy).toHaveBeenCalled();
+    expect(component.canCreateCollections).toBe(false);
+    expect(component.userIsLoggedIn).toBe(true);
+  }));
+
+  it('should stop the creation process if another' +
+    ' creation is in progress', () => {
+    component.canCreateCollections = false;
+    component.creationInProgress = false;
+
+    const explorationCreationServiceSpy = spyOn(
+      explorationCreationService, 'createNewExploration');
+
+    component.initCreationProcess();
+    expect(explorationCreationServiceSpy).toHaveBeenCalledTimes(1);
+
+    component.creationInProgress = true;
```

Comment:
Just to check, shouldn't this be an expectation (wouldn't it be expected to happen as a consequence of calling initCreationProcess())?

Reply:
Yep, changed it to an expectation